docs/integrations_api.yml

```diff
@@ -0,0 +1,135 @@
+swagger: '2.0'
+info:
+  version: '0.1'
+  title: User administration and authentication
+  description: |
+    An API for user administration and user authentication handling. Intended for use by the web GUI.
+
+basePath: '/api/integrations/0.1/useradm'
+host: 'docker.mender.io:8080'
+schemes:
+  - https
+
+paths:
+  /auth/login:
+    post:
+      summary: Log in to Mender
+      description: |
+        Accepts user credentials via standard Basic Auth, and returns a
+        JWT token to be used for authentication in subsequent requests.
+
+        When the backend is started for the first time (no users yet),
+        the credentials can be omitted. A special-purpose token will
+        be issued, allowing only the first-time user creation.
+      parameters:
+        - name: Authorization
+          in: header
+          description: |
+            Standard Basic Auth header, based on user's credentials.
+            If it's omitted - a valid token will be issued only when there are
+            no users defined yet in the database.
+          required: false
+          type: string
+      responses:
+        200:
+          description: |
+            Authentication successful - a new JWT is issued and returned.
+            The JWT is signed with the API's private key ('RS256' signing algorithm),
+            and contains the following standard claims:
+            * 'iss' - issuer
+            * 'exp' - expiry date
+            * 'sub' - unique, autogenerated user ID
+            * 'jti' - token's unique identifier (tracked for the purpose of revocation)
```

Comment:
do we plan to do token revocation? - it may make it harder to decentralise token verification in the future